### PR TITLE
Inject a TreeHasher rather than just a Hasher

### PR DESCRIPTION
**Pull request by Martin2112**
Addresses TODO. Moves dependency on RFC6962 hasher further up in the
system.

### CODE REVIEW

File: log/sequencer_test.go [hunk 1]
```diff
@@ -13,6 +13,7 @@ import (
 	"github.com/google/trillian/util"
 	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/assert"
+	"github.com/google/trillian/merkle"
```
Comment (AlCutter) [inline log/sequencer_test.go]: import ordering is wonky

File: server/sequencer_manager.go [hunk 2]
```diff
@@ -7,6 +7,7 @@ import (
 	"github.com/google/trillian"
 	"github.com/google/trillian/log"
 	"github.com/google/trillian/util"
+	"github.com/google/trillian/merkle"
```
Comment (AlCutter) [inline server/sequencer_manager.go]: import ordering

Comment (Martin2112) [inline server/sequencer_manager.go]: Done.

File: log/sequencer_test.go (hunk 1 above)
Comment (Martin2112) [inline log/sequencer_test.go]: Done.

**Pull request merged**